icu4c/source/i18n/calendar.cpp

```diff
@@ -870,7 +870,28 @@ Calendar::createInstance(const TimeZone& zone, UErrorCode& success)
 Calendar* U_EXPORT2
 Calendar::createInstance(const Locale& aLocale, UErrorCode& success)
 {
-    return createInstance(TimeZone::createDefault(), aLocale, success);
+    Calendar* res = createInstance(TimeZone::createDefault(), aLocale, success);
```

Comment:
Did you consider making a function that returns the locale's time zone if it is present, or `TimeZone::createDefault()` if it is not, instead of `adoptTimeZoneFromLocale`?  You could also use it in `smpdtfmt.cpp`.  For example:

```cpp
TimeZone* TimeZone::forLocaleOrDefault(const Locale& locale) {
  if (/* locale has time zone keyword */) {
    // return locale time zone
  } else {
    return TimeZone::createDefault();
  }
}
```

Reply:
DONE